src/dmd/mtype.d

```diff
@@ -6521,3 +6449,83 @@ const(char*)[2] toAutoQualChars(Type t1, Type t2)
     }
     return [s1, s2];
 }
+
+
+/**
+ * For each active modifier (MODFlags.const_, MODFlags.immutable_, etc) call `fp` with a
+ * void* for the work param and a string representation of the attribute.
+ */
+int modifiersApply(TypeFunction tf, void* param, int function(void*, const(char)[]) fp)
+{
+    immutable ubyte[4] modsArr = [MODFlags.const_, MODFlags.immutable_, MODFlags.wild, MODFlags.shared_];
+
+    foreach (modsarr; modsArr)
+    {
+        if (tf.mod & modsarr)
+        {
+            OutBuffer buf;
+            MODtoBuffer(&buf, modsarr);
+            const len = buf.offset;
+            if (int res = fp(param, buf.extractData()[0 .. len]))
```

Comment:
It still leaks memory, once each time through the loop. The strings produced by MODtoString are all literals and don't need to be duplicated.

Reply:
I mean MODtoBuffer.